crates/validation/src/order/state/order_validators.rs

```diff
@@ -6,15 +6,18 @@ use angstrom_types::{
 };
 use gas_set::EnsureGasSet;
 use max_gas_lt_min::EnsureMaxGasLessThanMinAmount;
+use price_set::EnsurePriceSet;
 
 pub mod amount_set;
 pub mod gas_set;
 pub mod max_gas_lt_min;
+pub mod price_set;
 
-pub const ORDER_VALIDATORS: [OrderValidator; 3] = [
+pub const ORDER_VALIDATORS: [OrderValidator; 4] = [
     OrderValidator::EnsureAmountSet(EnsureAmountSet),
     OrderValidator::EnsureMaxGasLessThanMinAmount(EnsureMaxGasLessThanMinAmount),
-    OrderValidator::EnsureGasSet(EnsureGasSet)
+    OrderValidator::EnsureGasSet(EnsureGasSet),
+    OrderValidator::EnsurePriceSet(EnsurePriceSet)
```

Comment:
order is wrong here, should be the ensures that don't do any math first for all of the zero checks and then last should be ensure max gas lt